### PR TITLE
Add book citations

### LINKED ISSUE #208: Add citation entries
Most R packages are cited only through links. We need to add formal citations.

### PR DESCRIPTION
**Pull request by emilyriederer**
This PR incorporates all package citations (will close #208). A few notes:

- My overall strategy was to ensure every package was cited in each section in which it appeared. The two exceptions are `knitr` and `rmarkdown` which I only cite on first appearance

- I did leave the links along with the citations in Chapter 10 (Tables) because here there seemed to be more of a directive (go read more here) than simply a citation. I notice some inconsistency here of what is linked to GitHub versus CRAN. I left as-is in case there was a deeper reason that I didn't know, but if you want me to standardize just let me know

- I did not cite package names only in biographies

- I left a list commented out in the references Rmd with the names of the GitHub and bio packages in case we would ever want to check later if these have been moved to CRAN or otherwise used. I can certainly delete if it's just clutter

### CODE REVIEW

**Comment by emilyriederer** [comment]
I should add for reviewing purposes: my methodology for finding all the packages to cite was essentially extracting all words wrapped in `**` throughout the book. Not very elegant but this got the job done:

```
files <- list.files()
chapters <- grep("\\d+-.*.Rmd$", files, value = TRUE)
contents <- lapply(chapters, FUN = function(x) readLines(x))
contents_all <- paste(unlist(contents), collapse = " ")
packages <- 
  stringr::str_extract_all(contents_all, "\\*\\*[A-Za-z0-9]*\\*\\*") %>%
  .[[1]] %>%
  unique() %>%
  gsub("\\*\\*", "", .)
dput(packages) %>% writeClipboard()
```

From there, I manually excluded LaTeX packages, GitHub-only R packages, and a few miscellaneous phrases that used the same formatting.

File: 18-references.Rmd [hunk 1]
```diff
@@ -7,12 +7,90 @@
...
+'sass',
+'roxygen2',
+'usethis',
+'rmdformats',
+'flexdashboard',
+'learnr',
+'rticles',
+'spelling',
+'googledrive',
+'workflowr'
```
Comment (cderv) [inline 18-references.Rmd:81]: Maybe we could put them into alphabetical order for easier checking in the future ? 

```suggestion
'animation',
'blastula',
'blogdown',
'bookdown',
'broom',
'Cairo',
'cairoDevice',
'condformat',
'dagitty',
'diagram',
'DiagrammeR',
'distill',
'downloadthis',
'DT',
'flair',
'flexdashboard',
'flextable',
'formatR',
'formattable',
'gganimate',
'ggdag',
'ggplot2',
'gifski',
'googledrive',
'govdown',
'gt',
'gtsummary',
'huxtable',
'kableExtra',
'knitcitations',
'knitr',
'learnr',
'magick',
'nomnoml',
'officedown',
'officer',
'pagedown',
'pander',
'pixiedust',
'pkgdown',
'printr',
'r2d3',
'reactable',
'remedy',
'reticulate',
'revealjs',
'rgl',
'rhandsontable',
'rmarkdown',
'rmdformats',
'roxygen2',
'rsconnect',
'rticles',
'sass',
'spelling',
'stargazer',
'styler',
'svglite',
'tables',
'tangram',
'tikzDevice',
'tinytex',
'tufte',
'usethis',
'webshot',
'workflowr',
'xaringan',
'xfun',
'xtable',
'yaml',
'ztable'
```

Comment (cderv) [inline 18-references.Rmd:81]: Maybe we could put them into alphabetical order for easier checking in the future ? 

```suggestion
'animation',
'blastula',
'blogdown',
'bookdown',
'broom',
'Cairo',
'cairoDevice',
'condformat',
'dagitty',
'diagram',
'DiagrammeR',
'distill',
'downloadthis',
'DT',
'flair',
'flexdashboard',
'flextable',
'formatR',
'formattable',
'gganimate',
'ggdag',
'ggplot2',
'gifski',
'googledrive',
'govdown',
'gt',
'gtsummary',
'huxtable',
'kableExtra',
'knitcitations',
'knitr',
'learnr',
'magick',
'nomnoml',
'officedown',
'officer',
'pagedown',
'pander',
'pixiedust',
'pkgdown',
'printr',
'r2d3',
'reactable',
'remedy',
'reticulate',
'revealjs',
'rgl',
'rhandsontable',
'rmarkdown',
'rmdformats',
'roxygen2',
'rsconnect',
'rticles',
'sass',
'spelling',
'stargazer',
'styler',
'svglite',
'tables',
'tangram',
'tikzDevice',
'tinytex',
'tufte',
'usethis',
'webshot',
'workflowr',
'xaringan',
'xfun',
'xtable',
'yaml',
'ztable'
```

Comment (cderv) [inline 18-references.Rmd:81]: Maybe we could put them into alphabetical order for easier checking in the future ? 

```suggestion
'animation',
'blastula',
'blogdown',
'bookdown',
'broom',
'Cairo',
'cairoDevice',
'condformat',
'dagitty',
'diagram',
'DiagrammeR',
'distill',
'downloadthis',
'DT',
'flair',
'flexdashboard',
'flextable',
'formatR',
'formattable',
'gganimate',
'ggdag',
'ggplot2',
'gifski',
'googledrive',
'govdown',
'gt',
'gtsummary',
'huxtable',
'kableExtra',
'knitcitations',
'knitr',
'learnr',
'magick',
'nomnoml',
'officedown',
'officer',
'pagedown',
'pander',
'pixiedust',
'pkgdown',
'printr',
'r2d3',
'reactable',
'remedy',
'reticulate',
'revealjs',
'rgl',
'rhandsontable',
'rmarkdown',
'rmdformats',
'roxygen2',
'rsconnect',
'rticles',
'sass',
'spelling',
'stargazer',
'styler',
'svglite',
'tables',
'tangram',
'tikzDevice',
'tinytex',
'tufte',
'usethis',
'webshot',
'workflowr',
'xaringan',
'xfun',
'xtable',
'yaml',
'ztable'
```

**Comment by emilyriederer** [comment]
Thanks for the changes, @cderv ! I merged in the alphabetical order and fixed `blastula`. 

That's a good point about `::`. In my mind, if we weren't mentioning a package name, we weren't exactly "referencing" it so I hadn't included those. I can definitely add those in if I should. Please let me know! Here is the list of additional packages that would be captured with that method and a brief description of how they are used:

- remotes: used in instructions for GitHub package installations
- jsonlite: used in code chunk in #lua-filter section but not described 
- htmltools: mentioned twice in the #include-html tip
- tools: mentioned in an inline piece of code in the #cache section (`cache.extra = tools::md5sum('my-precious.csv')`)
- codetools: mentioned in the cache invalidation section in way of explanation as opposed to recommendation ("global variables are identified by `codetools::findGlobals()` when `hash = "auto"`")
- base: Used to explain that the `cat` engine uses `base::cat`
- tibble: used in non-echoed code chunk to make a table of RStudio shortcuts

File: 04-content.Rmd [hunk 2]
```diff
@@ -79,7 +79,7 @@ We used the title as the example in this section, but the idea can be applied to
...
+When a Rmd document is compiled, all of its metadata in the YAML section will be stored in the list object `rmarkdown::metadata`. For example, `rmarkdown::metadata$title` gives you the title of the document. You can use this `metadata` object in your R code, so that you do not need to hard-code information that has been provided in the YAML metadata. For example, when you send an email with [the **blastula** package](https://github.com/rich-iannone/blastula) [@R-blastula] within an Rmd document, you may use the title of the document as the email subject, and get the sender information from the author field:
```
[the remark this answers is not in the document]
Comment (emilyriederer) [reply 04-content.Rmd]: Good catch - thank you!

**Comment by cderv** [comment]
I don't know the good practice on those package used with `::`. I just pointed out the question 😄 Thanks for the list @emilyriederer ! 

@yihui you can decide and if nothing to do, you can merged. 👍

**Pull request merged**